src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/RefactorComponentCreateParams.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
+{
+    internal sealed class RefactorComponentCreateParams
+    {
+        public Uri Uri { get; set; }
+        public string Name { get; set; }
+        public string Where { get; set;  }
```

Comment:
```suggestion
        public string Path { get; set; }
```

Reply:
Also, is there a reason for this being passed in by the provider rather than the resolver figuring out the creation path based on `Uri` and `Name`?

Reply:
Mainly for the sake of extensibility. Perhaps down the road we'll have another reason to create a component; having a way to arbitrarily place a generated component anywhere might come in handy. I can change it if there's no need!

Reply:
What you have looks good but I asked because right now the logic is somewhat split. The extraction of namespace happens in the resolver whereas figuring out the new component path happens in the provider. Having said that, I don't feel strongly about changing this one way or the other. So let's leave this as is.